src/services/patternMatcher.ts

```diff
@@ -0,0 +1,810 @@
+        isDottedPattern: boolean;
+    }
+
+    // First we break up the pattern given by dots.  Each portion of the pattern between the
+    // dots is a 'Segment'.  The 'Segment' contains information about the entire section of 
+    // text between the dots, as well as information about any individual 'Words' that we 
+    // can break the segment into.  A 'Word' is simply a contiguous sequence of characters
+    // that can appear in a typescript identifier.  So "GetKeyword" would be one word, while
+    // "Get Keyword" would be two words.  Once we have the individual 'words', we break those
+    // into constituent 'character spans' of interest.  For example, while 'UIElement' is one
+    // word, it make character spans corresponding to "U", "I" and "Element".  These spans
+    // are then used when doing camel cased matches against candidate patterns.
+    interface Segment {
+        // Information about the entire piece of text between the dots.  For example, if the 
+        // text between the dots is 'GetKeyword', then TotalTextChunk.Text will be 'GetKeyword' and 
+        // TotalTextChunk.CharacterSpans will correspond to 'Get', 'Keyword'.
+        totalTextChunk: TextChunk;
+
+        // Information about the subwords compromising the total word.  For example, if the 
+        // text between the dots is 'GetFoo KeywordBar', then the subwords will be 'GetFoo' 
+        // and 'KeywordBar'.  Those individual words will have CharacterSpans of ('Get' and 
+        // 'Foo') and('Keyword' and 'Bar') respectively. 
+        subWordTextChunks: TextChunk[];
+    }
+
+    // Information about a chunk of text from the pattern.  The chunk is a piece of text, with 
+    // cached information about the character spans within in.  Character spans are used for 
+    // camel case matching.
+    interface TextChunk {
+        // The text of the chunk.  This should be a contiguous sequence of character that could
+        // occur in a symbol name.
+        text: string;
+
+        // The text of a chunk in lower case.  Cached because it is needed often to check for
+        // case insensitive matches.
+        textLowerCase: string;
+
+        // Whether or not this chunk is entirely lowercase. We have different rules when searching
+        // for something entirely lowercase or not.
+        isLowerCase: boolean;
+
+        // The spans in this text chunk that we think are of interest and should be matched 
+        // independently.  For example, if the chunk is for "UIElement" the the spans of interest
+        // correspond to "U", "I" and "Element".  If "UIElement" isn't found as an exaxt, prefix.
+        // or substring match, then the character spans will be used to attempt a camel case match.
+        characterSpans: TextSpan[];
+    }
+
+    function createPatternMatch(kind: PatternMatchKind, punctuationStripped: boolean, isCaseSensitive: boolean, camelCaseWeight?: number): PatternMatch {

```

Comment:
I would just inline the object literals instead. This is not an interesting function.